src/parser/xlsx/xlsx-parser.ts

```diff
@@ -0,0 +1,164 @@
+import * as xlsx from 'xlsx';
+import { SheetNamesEnum } from '../../constants/sheet-names';
+import { LearningObjectSheetColumnNames } from '../../constants/learning-object-sheet-column-names';
+import schemaValidator from './utilities/schema-validator.util';
+import AmdSchema from '../../schema/amd-schema';
+import Lo from '../../model/lo';
+import Amd from '../../model/amd';
+import Screen from '../../model/screen';
+import TransformError from '../../utilities/transform-error.util';
+import ApplicationErrors from '../../utilities/application-errors';
+
+
+export default class XlsxParser implements Base {
+    public workbook: any;
+
+    constructor() { }
+
+    /**
+     * Parse the excel file, validate sheet names and column names then create the AMD object
+     * @param fileData 
+     * @returns {Amd} Returns AMD object
+     */
+
+    public parse(fileData: any) {
+        try {
+            this.workbook = xlsx.read(fileData, { type: 'buffer' });
+
+            // validate sheet names
+            if (!this.validateSheetNames()) {
+                throw new TransformError(ApplicationErrors.AMD_SHEET_NAMES_VALIDATION_FAILED);
+            }
+
+            // validate column names in learning object sheet
+            if (!this.validateColumnNames(this.workbook.Sheets[SheetNamesEnum.LEARNING_OBJECT], LearningObjectSheetColumnNames)) {
+                throw new TransformError(ApplicationErrors.LEARNING_OBJECT_SHEET_COLUMN_NAMES_VALIDATION_FAILED);
+            }
+
+            return this.parseLearningObjectSheet();
+        }
+        catch (error) {
+            throw error;
+        }
+    }
+
+    /**
+     * Checks all required sheets are present in excel file or not
+     * @returns Returns boolean value for whether all required sheets are present in excel file or not
+     */
+    public validateSheetNames() {
+        return Object.values(SheetNamesEnum).every((knownSheetName) => this.workbook.SheetNames.find((availableSheet: string) => availableSheet?.trim() === knownSheetName?.trim()));
+    }
+
+    /**
+     * Checks if all required column are present in the input sheet or not
+     * @param sheet 
+     * @param {string[]} knownColumnNames 
+     * @returns Returns boolean value for whether all required column are present in the input sheet or not
+     */
+    public validateColumnNames(sheet: any, knownColumnNames: string[]) {
```

Comment:
Please rename it to validateLOSheetColumns

Reply:
validateSheetColumns